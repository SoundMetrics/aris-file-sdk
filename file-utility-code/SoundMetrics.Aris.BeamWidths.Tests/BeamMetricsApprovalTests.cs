using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using SoundMetrics.Aris.Headers;
using System.Text;

namespace SoundMetrics.Aris.BeamWidths.Tests
{
    [TestClass]
    [UseApprovalSubdirectory("approval-files"), UseReporter(typeof(DiffReporter))]
    public sealed class BeamMetricsApprovalTests
    {
        [TestMethod]
        public void BeamWidths_ARIS3000_128()
            => RunTest(new BeamConfig(ArisSystemType.Aris3000, 128, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS3000_64()
            => RunTest(new BeamConfig(ArisSystemType.Aris3000, 64, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1200_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1800_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 48, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1800_96()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 96, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1200_Telephoto_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.Telephoto));

        [TestMethod]
        public void BeamWidths_ARIS1800_Telephoto_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.Telephoto));

        [TestMethod]
        public void BeamWidths_ARIS1800_Telephoto_96()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 96, LensType.Telephoto));

        private static void RunTest(BeamConfig cfg)
        {
            var stringOutput = BuildStringOutput(cfg);
            Approvals.Verify(stringOutput);
        }

        private static string BuildStringOutput(BeamConfig cfg)
        {
            var beamInfos = Metrics.GetBeamInformation(cfg);
            var stringOutput = BuildOutput(beamInfos);
            return stringOutput;

            static string BuildOutput(BeamInfo[] beamInfos)
            {
                var buffer = new StringBuilder();
                buffer.AppendLine($"# of BeamInfo members: [{beamInfos.Length}]");

                foreach (var beamInfo in beamInfos)
                {
                    buffer.AppendLine(beamInfo.ToString());
                }

                return buffer.ToString();
            }
        }
    }
}
