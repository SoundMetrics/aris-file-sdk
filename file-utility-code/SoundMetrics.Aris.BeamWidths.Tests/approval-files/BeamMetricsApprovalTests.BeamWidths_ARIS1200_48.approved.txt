# of BeamInfo members: [48]
BeamInfo { BeamNumber = 0, Center = -13.5735, Left = -13.7893, Right = -13.3577 }
BeamInfo { BeamNumber = 1, Center = -13.1419, Left = -13.3577, Right = -12.9211 }
BeamInfo { BeamNumber = 2, Center = -12.7003, Left = -12.9211, Right = -12.4723 }
BeamInfo { BeamNumber = 3, Center = -12.2442, Left = -12.4723, Right = -12.0082 }
BeamInfo { BeamNumber = 4, Center = -11.7721, Left = -12.0082, Right = -11.5301 }
BeamInfo { BeamNumber = 5, Center = -11.2881, Left = -11.5301, Right = -11.0401 }
BeamInfo { BeamNumber = 6, Center = -10.7921, Left = -11.0401, Right = -10.5373 }
BeamInfo { BeamNumber = 7, Center = -10.2824, Left = -10.5373, Right = -10.0211 }
BeamInfo { BeamNumber = 8, Center = -9.7597, Left = -10.0211, Right = -9.4902 }
BeamInfo { BeamNumber = 9, Center = -9.2206, Left = -9.4902, Right = -8.9418 }
BeamInfo { BeamNumber = 10, Center = -8.663, Left = -8.9418, Right = -8.3765 }
BeamInfo { BeamNumber = 11, Center = -8.0899, Left = -8.3765, Right = -7.7963 }
BeamInfo { BeamNumber = 12, Center = -7.5026, Left = -7.7963, Right = -7.2023 }
BeamInfo { BeamNumber = 13, Center = -6.9019, Left = -7.2023, Right = -6.5958 }
BeamInfo { BeamNumber = 14, Center = -6.2897, Left = -6.5958, Right = -5.9779 }
BeamInfo { BeamNumber = 15, Center = -5.6661, Left = -5.9779, Right = -5.3476 }
BeamInfo { BeamNumber = 16, Center = -5.0291, Left = -5.3476, Right = -4.7057 }
BeamInfo { BeamNumber = 17, Center = -4.3823, Left = -4.7057, Right = -4.0544 }
BeamInfo { BeamNumber = 18, Center = -3.7264, Left = -4.0544, Right = -3.3948 }
BeamInfo { BeamNumber = 19, Center = -3.0632, Left = -3.3948, Right = -2.7282 }
BeamInfo { BeamNumber = 20, Center = -2.3931, Left = -2.7282, Right = -2.0551 }
BeamInfo { BeamNumber = 21, Center = -1.717, Left = -2.0551, Right = -1.3757 }
BeamInfo { BeamNumber = 22, Center = -1.0343, Left = -1.3757, Right = -0.6902 }
BeamInfo { BeamNumber = 23, Center = -0.3461, Left = -0.6902, Right = 0 }
BeamInfo { BeamNumber = 24, Center = 0.3461, Left = 0, Right = 0.6902 }
BeamInfo { BeamNumber = 25, Center = 1.0343, Left = 0.6902, Right = 1.3757 }
BeamInfo { BeamNumber = 26, Center = 1.717, Left = 1.3757, Right = 2.0551 }
BeamInfo { BeamNumber = 27, Center = 2.3931, Left = 2.0551, Right = 2.7282 }
BeamInfo { BeamNumber = 28, Center = 3.0632, Left = 2.7282, Right = 3.3948 }
BeamInfo { BeamNumber = 29, Center = 3.7264, Left = 3.3948, Right = 4.0544 }
BeamInfo { BeamNumber = 30, Center = 4.3823, Left = 4.0544, Right = 4.7057 }
BeamInfo { BeamNumber = 31, Center = 5.0291, Left = 4.7057, Right = 5.3476 }
BeamInfo { BeamNumber = 32, Center = 5.6661, Left = 5.3476, Right = 5.9779 }
BeamInfo { BeamNumber = 33, Center = 6.2897, Left = 5.9779, Right = 6.5958 }
BeamInfo { BeamNumber = 34, Center = 6.9019, Left = 6.5958, Right = 7.2023 }
BeamInfo { BeamNumber = 35, Center = 7.5026, Left = 7.2023, Right = 7.7963 }
BeamInfo { BeamNumber = 36, Center = 8.0899, Left = 7.7963, Right = 8.3765 }
BeamInfo { BeamNumber = 37, Center = 8.663, Left = 8.3765, Right = 8.9418 }
BeamInfo { BeamNumber = 38, Center = 9.2206, Left = 8.9418, Right = 9.4902 }
BeamInfo { BeamNumber = 39, Center = 9.7597, Left = 9.4902, Right = 10.0211 }
BeamInfo { BeamNumber = 40, Center = 10.2824, Left = 10.0211, Right = 10.5373 }
BeamInfo { BeamNumber = 41, Center = 10.7921, Left = 10.5373, Right = 11.0401 }
BeamInfo { BeamNumber = 42, Center = 11.2881, Left = 11.0401, Right = 11.5301 }
BeamInfo { BeamNumber = 43, Center = 11.7721, Left = 11.5301, Right = 12.0082 }
BeamInfo { BeamNumber = 44, Center = 12.2442, Left = 12.0082, Right = 12.4723 }
BeamInfo { BeamNumber = 45, Center = 12.7003, Left = 12.4723, Right = 12.9211 }
BeamInfo { BeamNumber = 46, Center = 13.1419, Left = 12.9211, Right = 13.3577 }
BeamInfo { BeamNumber = 47, Center = 13.5735, Left = 13.3577, Right = 13.7893 }
